refactor(cdr-tooltip, cdr-popup, cdr-form-group, cdr-popover): <scrip…

KenjiCrosland opened the pull request:
…t setup> and Vue 3 refactor

Refactor CdrPopup and Popover to use <script setup> syntax. Fix autoposition issue with popup.
Refactor addHandlers() method to fix breaking issue with refactored popup. Add UniqueiId to form
groups

mhewson commented:
:+1: ApPrOvE -- lets get this in before flex and run the visual regression

mhewson commented:
mhewson@C02NT179G3QP ~/code/rei-cedar (update-deps)$ gulp
/Users/mhewson/code/rei-cedar/gulpfile.js:355
s:test:post-clean', [ 'js:test:qunit' ], function() { return del( [ `${ PATHS.
                                                                    ^
SyntaxError: Unexpected token ILLEGAL
    at exports.runInThisContext (vm.js:73:16)
    at Module._compile (module.js:443:25)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Liftoff.handleArguments (/usr/local/lib/node_modules/gulp/bin/gulp.js:116:3)
    at Liftoff.<anonymous> (/usr/local/lib/node_modules/gulp/node_modules/liftoff/index.js:192:16)
    at module.exports (/usr/local/lib/node_modules/gulp/node_modules/liftoff/node_modules/flagged-respawn/index.js:17:3)

wjramos commented:
Likely related to version of node. I can change this from template strings to regular concatenation

wjramos commented:
Yup -- Template strings and other es6 features are supported in node 4+ (I assume you are using 0.12), I will update this for more backwards compatibility between versions of node

mhewson commented:
brew upgrade node - working now
maybe a note in the readme on node vision requirements?

wjramos commented:
I removed the template strings, so it should be more backwards compatible

mhewson commented:
getting a fail on npm run test-ui

On Mon, Mar 7, 2016 at 11:05 AM, Jordan Ramos <notifications@github.com>
wrote:

> I removed the template strings, so it should be more backwards compatible
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rei/rei-cedar/pull/47#issuecomment-193398909>.
>

wjramos commented:
@mhewson Good to merge?

wjramos commented:
@mhewson we good?

mhewson commented:
yes sir

Commit:
refactor(cdr-tooltip, cdr-popup, cdr-form-group, cdr-popover): <script setup> and Vue 3 refactor

Refactor CdrPopup and Popover to use <script setup> syntax. Fix autoposition issue with popup.
Refactor addHandlers() method to fix breaking issue with refactored popup. Add uniqueid to form
groups

Commit:
fix(cdr-popover): fix double firing of open event

Wrote a check to ensure that the "open" event on the popover doesn't fire if it's already open
